sonic-xcvrd/xcvrd/sff_mgr.py

```diff
@@ -0,0 +1,407 @@
+
+    def __init__(self, namespaces, main_thread_stop_event, platform_chassis, helper_logger):
+        threading.Thread.__init__(self)
+        self.name = "SffManagerTask"
+        self.exc = None
+        self.task_stopping_event = threading.Event()
+        self.main_thread_stop_event = main_thread_stop_event
+        self.helper_logger = helper_logger
+        self.platform_chassis = platform_chassis
+        # port_dict holds data per port entry with logical_port_name as key, it
+        # maintains local copy of the following DB fields:
+        #   CONFIG_DB PORT_TABLE 'index', 'channel', 'admin_status'
+        #   STATE_DB PORT_TABLE 'host_tx_ready'
+        #   STATE_DB TRANSCEIVER_INFO 'type'
+        # plus 'asic_id' from PortChangeEvent.asic_id (asic_id always gets
+        # filled in handle_port_update_event function based on asic_context)
+        # Its port entry will get deleted upon CONFIG_DB PORT_TABLE DEL.
+        # Port entry's 'type' field will get deleted upon STATE_DB TRANSCEIVER_INFO DEL.
+        self.port_dict = {}
+        # port_dict snapshot captured in the previous event update loop
+        self.port_dict_prev = {}
+        self.xcvr_table_helper = XcvrTableHelper(namespaces)
+        self.namespaces = namespaces
+
+    def log_notice(self, message):
+        self.helper_logger.log_notice("SFF: {}".format(message))
+
+    def log_warning(self, message):
+        self.helper_logger.log_warning("SFF: {}".format(message))
+
+    def log_error(self, message):
+        self.helper_logger.log_error("SFF: {}".format(message))
+
+    def on_port_update_event(self, port_change_event):
+        if (port_change_event.event_type
+                not in [port_change_event.PORT_SET, port_change_event.PORT_DEL]):
+            return
+
+        lport = port_change_event.port_name
+        pport = port_change_event.port_index
+        asic_id = port_change_event.asic_id
+
+        # Skip if it's not a physical port
+        if not lport.startswith('Ethernet'):
+            return
+
+        # Skip if the physical index is not available
+        if pport is None:
+            return
+
+        if port_change_event.port_dict is None:
+            return
+
+        if port_change_event.event_type == port_change_event.PORT_SET:
+            if lport not in self.port_dict:
+                self.port_dict[lport] = {}
+            if pport >= 0:
+                self.port_dict[lport]['index'] = pport
+            # This field comes from CONFIG_DB PORT_TABLE. This is the channel
+            # that blongs to this logical port, 0 means all channels. tx_disable
+            # API needs to know which channels to disable/enable for a
+            # particular physical port.
+            if 'channel' in port_change_event.port_dict:
+                self.port_dict[lport]['channel'] = port_change_event.port_dict['channel']
+            # This field comes from STATE_DB PORT_TABLE
+            if 'host_tx_ready' in port_change_event.port_dict:
```

Comment:
Why not use constants for 'host_tx_ready' and 'admin_status' instead of string literals? These strings are commonly used in sff_mgr.py.